src/Support/SelectFields.php

```diff
@@ -214,6 +220,10 @@ protected static function handleFields(
                             if (! array_key_exists($foreignKey, $field)) {
                                 $field['fields'][$foreignKey] = self::FOREIGN_KEY;
                             }
+
+                            if (method_exists($relation, 'getMorphType')) {
```

Comment:
Shouldn't this rather use `is_a` like in the if condition a few lines above (for the two morph types entering this condition)?